integrationtests/gitjob/controller/controller_test.go

```diff
@@ -616,6 +652,39 @@ var _ = Describe("GitJob controller", func() {
 				}, "30s", "1s").Should(Equal(stableCommit))
 			})
 		})
+
+		When("WebhookCommit changes and user clicks ForceUpdateGeneration", func() {
```

Comment:
nit: a click is specific to the Rancher UI, but generation update may be forced through an API or `kubectl` call.
```suggestion
		When("WebhookCommit changes and user forces a redeployment", func() {
```